osu.Game.Rulesets.Osu/Difficulty/Evaluators/AimEvaluator.cs

```diff
@@ -112,18 +121,48 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current, bool with
                 velocityChangeBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuLastObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuLastObj.StrainTime), 2);
             }
 
-            if (osuLastObj.BaseObject is Slider)
+            // Add in acute angle bonus or wide angle bonus + velocity change bonus, whichever is larger.
+            aimStrain += Math.Max(acuteAngleBonus * acute_angle_multiplier, wideAngleBonus * wide_angle_multiplier + velocityChangeBonus * velocity_change_multiplier);
+
+            double sliderJumpBonus = 0;
+            if (osuLastObj.BaseObject is Slider slider)
             {
-                // Reward sliders based on velocity.
-                sliderBonus = osuLastObj.TravelDistance / osuLastObj.TravelTime;
+                // Take just slider heads into account because we're computing sliderjumps, not slideraim
+                sliderJumpBonus = slider_jump_multiplier * aimStrain * sliderJumpsAdjustingMultiplier;
+
+                // Reward more if sliders and circles are alternating (actually it's still lower than several sliders in a row)
+                if (osuLastLastObj.BaseObject is HitCircle)
+                {
+                    double alternatingBonus = slider_jump_multiplier * osuLastObj.JumpDistance / osuLastObj.StrainTime;
+
+                    if (osuLastObj.StrainTime > osuLastLastObj.StrainTime)
+                        alternatingBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuLastObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuLastObj.StrainTime), 2);
+
+                    sliderJumpBonus += alternatingBonus;
+                }
+
+                // If slider was slower than notes before - punish it
+                if (osuCurrObj.StrainTime > osuLastObj.StrainTime)
+                    sliderJumpBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuLastObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuLastObj.StrainTime), 2);
+
+                // Punish too short sliders to prevent cheesing (cheesing is still possible, but it's very rare)
+                double sliderLength = slider.Velocity * slider.SpanDuration;
+                if (sliderLength < slider.Radius)
+                    sliderJumpBonus *= sliderLength / slider.Radius;
             }
```

Comment:
rename this variable to `sliderDuration` as length can ambiguously be understood to mean distance here.

i would be expecting something to do with follow circle radius here (2.4x). did you just not find that to be necessary? your comment does state that cheesing still occurs with this code, so something to consider

Reply:
sliderjumps is not connected with sliderbody difficulty in any way, so there's no sense in using follow circle radius
cheesing can occur cuz you can hide long sliderbody under the circle, so the slider will appear like a circle
but i haven't seen it in ranked maps so it's good

Reply:
also, it's indeed length, not a duration
do you misunderstanding something?
velocity * time = distance
velocity * duration = length

Reply:
> sliderjumps is not connected with sliderbody difficulty in any way, so there's no sense in using follow circle radius

but slider leniency will allow short enough sliders to effectively allow the movement as if it were circles. mappers have used this technique for over a decade at this point

> also, it's indeed length, not a duration
yeah I think I just get tripped up by people using the term "length of time" in sentences. duration is just less ambiguous IMO

Reply:
> but slider leniency will allow short enough sliders to effectively allow the movement as if it were circles. mappers have used this technique for over a decade at this point

this is literally the point of sliderjumps
even tho technically they allow the same movement - they're still harder because they're harder to read

> yeah I think I just get tripped up by people using the term "length of time" in sentences. duration is just less ambiguous IMO

how duration is less ambiguous if duration literally means time, when this is clearly length in pixels

Reply:
... i think i confused myself on the duration and length thing again. i just read myself and i am not making any sense, disregard